server/src/main/java/org/opensearch/index/shard/CheckpointRefreshListener.java

```diff
@@ -21,14 +21,15 @@
  *
  * @opensearch.internal
  */
-public class CheckpointRefreshListener implements ReferenceManager.RefreshListener {
+public class CheckpointRefreshListener extends CloseableRetryableRefreshListener {
 
     protected static Logger logger = LogManager.getLogger(CheckpointRefreshListener.class);
 
     private final IndexShard shard;
     private final SegmentReplicationCheckpointPublisher publisher;
 
     public CheckpointRefreshListener(IndexShard shard, SegmentReplicationCheckpointPublisher publisher) {
+        super(null);
```

Comment:
nit: Can we add a default constructor to parent class?

Reply:
Addressed.